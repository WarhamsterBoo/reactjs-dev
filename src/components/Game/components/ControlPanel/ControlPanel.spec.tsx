import { mount, shallow } from "enzyme";
import React from "react";
import { ControlPanel } from ".";

describe("ControlPanel", () => {
  it("should render", () => {
    const sut = shallow(<ControlPanel onControlButtonClick={jest.fn()} />);

    expect(sut).toMatchSnapshot();
  });

  it.each`
    buttonId    | action
    ${"stop"}   | ${"stop"}
    ${"run"}    | ${"run"}
    ${"pause"}  | ${"pause"}
    ${"slower"} | ${"slower"}
    ${"normal"} | ${"normal"}
    ${"faster"} | ${"faster"}
  `(
    "should call onClick callback with action '$action' for button with Id '$buttonId'",
    ({ buttonId, action }) => {
      const fakeOnClick = jest.fn();
      const sut = mount(<ControlPanel onControlButtonClick={fakeOnClick} />);

      sut.find(`button[id="${buttonId}"]`).simulate("click");

      expect(fakeOnClick).toBeCalledTimes(1);
      expect(fakeOnClick).toHaveBeenCalledWith(action);
    }
  );
});
