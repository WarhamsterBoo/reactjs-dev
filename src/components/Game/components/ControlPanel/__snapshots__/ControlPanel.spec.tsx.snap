// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlPanel should render 1`] = `
<Styled(div)>
  <Styled(button)
    id="stop"
    onClick={[Function]}
  >
    Stop
  </Styled(button)>
  <Styled(button)
    id="run"
    onClick={[Function]}
  >
    Run
  </Styled(button)>
  <Styled(button)
    id="pause"
    onClick={[Function]}
  >
    Pause
  </Styled(button)>
  <Styled(button)
    id="faster"
    onClick={[Function]}
  >
    Faster
  </Styled(button)>
  <Styled(button)
    id="normal"
    onClick={[Function]}
  >
    Normal
  </Styled(button)>
  <Styled(button)
    id="slower"
    onClick={[Function]}
  >
    Slower
  </Styled(button)>
</Styled(div)>
`;
